scripts/west_commands/fetchers/__init__.py

```diff
@@ -0,0 +1,45 @@
+# Copyright (c) 2022 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import importlib
+import logging
+import os
+from pathlib import Path
+
+from fetchers.core import ZephyrBlobFetcher
+
+_logger = logging.getLogger('fetchers')
+
+def _import_fetcher_module(fetcher_name):
+    try:
+        importlib.import_module(f'fetchers.{fetcher_name}')
+    except ImportError as ie:
+        # Fetchers are supposed to gracefully handle failures when they
+        # import anything outside of stdlib, but they sometimes do
+        # not. Catch ImportError to handle this.
+        _logger.warning(f'The module for fetcher "{fetcher_name}" '
+                        f'could not be imported ({ie}). This most likely '
+                        'means it is not handling its dependencies properly. '
+                        'Please report this to the zephyr developers.')
+
+# We import these here to ensure the BlobFetcher subclasses are
+# defined; otherwise, BlobFetcher.get_fetchers() won't work.
+
+# Those do not contain subclasses of ZephyrBlobFetcher
+name_blocklist = ['__init__', 'core']
+
+fetchers_dir = Path(__file__).parent.resolve()
+for f in [f for f in os.listdir(fetchers_dir)]:
```

Comment:
Unsure why @mbolivar-nordic did not do the same in the runners module:
https://github.com/zephyrproject-rtos/zephyr/blob/main/scripts/west_commands/runners/__init__.py#L54

But I personally prefer this approach to the list of names that needs to be maintained.

Reply:
I guess it's really a question of allowlist (the runners approach) and blocklist (the fetchers approach). Either one works and I guess it's up to taste to some extent.

Reply:
Makes sense, thanks for the insight.

Reply:
Did you really mean to make this a separate commit in the same PR? Why?

Reply:
Yes, I did this intentionally. I thought it was valuable to keep a record of my thought process when creating the infrastructure.